.github/workflows/ibis-backends.yml

```diff
@@ -9,6 +9,10 @@ on:
     branches:
       - master
 
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
```

Comment:
Are the semantics here to cancel any in progress builds when a new build shows up that:

1. Is in the ibis-project/ibis (or `$GITHUB_USER/ibis`) repo
2. Has the same head ref or SHA (the latter seems unlikely given SHA256)
3. Is the same workflow

Is my understanding correct?

Reply:
Yes. See additional explanation about the `${{ github.head_ref || github.sha }}` part here https://github.com/apache/arrow/pull/11520#issuecomment-952978639